compiler/rustc_hir/src/hir.rs

```diff
@@ -2607,13 +2607,20 @@ pub struct ImplItemRef<'hir> {
     pub defaultness: Defaultness,
 }
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash, Encodable, Debug, HashStable_Generic)]
+#[derive(Copy, Clone, Hash, Encodable, Debug, HashStable_Generic)]
 pub enum AssocItemKind {
     Const,
     Fn { has_self: bool },
     Type,
 }
 
+impl PartialEq for AssocItemKind {
```

Comment:
Deriving Hash but having a custom PartialEq will lead to bugs: https://doc.rust-lang.org/std/collections/struct.HashMap.html#. I think you need to also manually implement Hash.